File: src/main/java/com/acme/GreetingResource.java
package com.acme;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/greet")
public class GreetingResource {

    private final GraalPyService svc;

    @Autowired
    public GreetingResource(GraalPyService svc) {
        this.svc = svc;
    }

    @GetMapping(produces = MediaType.TEXT_XML_VALUE)
    public String greet() {
        return svc.greet("Spring Boot");
    }
}

File: src/main/kotlin/com/acme/GreetingResource.kt
package com.acme

import jakarta.inject.Inject
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces
import jakarta.ws.rs.core.MediaType

@Path("/greet")
class GreetingResource @Inject constructor(
    private val svc: GraalPyService
) {
    @GET
    @Produces(MediaType.TEXT_XML)
    fun greet() = svc.greet("Quarkus")
}
